climada/util/coordinates.py

```diff
@@ -1822,6 +1821,39 @@ def _read_raster_reproject(src, src_crs, dst_meta, band=None, geometry=None, dst
 
     return intensity
 
+def _add_gdal_vsi_prefix(path):
+    """Add one of GDAL's virtual file system prefixes if applicable
+
+    GDAL (and thus, rasterio) can be told to read data from compressed files without extracting the
+    data on disk ("GDAL Virtual File Systems"). This utility function tries to guess from a file's
+    suffix whether the file is compressed. If applicable, a prefix is added to the file's path that
+    tells GDAL to use the virtual file system feature.
+
+    For more information about the GDAL Virtual File Systems feature, see:
+    https://gdal.org/user/virtual_file_systems.html
+
+    Parameters
+    ----------
+    path : str or Path
+        Path to a (compressed) raster file to be opened with rasterio.
+
+    Returns
+    -------
+    path : str
+        The path with prefix if applicable, and the original path otherwise. This will always be
+        a string even if the path was provided as a Path object.
+    """
+    supported_suffixes = {
+        ".gz": "gzip",
+        ".zip": "zip",
+        ".tar": "tar",
+        ".tgz": "tar",
+    }
+    suffix = Path(path).suffix
+    if suffix in supported_suffixes:
+        path = f"/vsi{supported_suffixes[suffix]}/{path}"
+    return str(path)
```

Comment:
Previously, the prefixing was manually applied for `gz`-files in three places. But we can easily add support for other archive formats and we can avoid redundant code by introducing the helper function.

Note that the prefix "looks like a path". Since the result with the added prefix is not really a path, I would not convert it to a `Path` object. And for consistency, I would also convert to a `str` in the cases where no prefix is added at all.